### ISSUE TITLE
Zsync not working?

### ISSUE THREAD

**Issue by vinyfc93** [neovim/neovim#9893]
Hello! I'm trying to use zsync to download the latest version of neovim:
```sh
$ zsync2 https://github.com/neovim/neovim/releases/download/v0.3.4/nvim.appimage.zsync
```
and receive this error:
```sh
zsync2 version 2.0.0-alpha-1 (commit bacc238), build 137 built on 2019-01-04 05:09:05 UTC
Checking for changes...
Cannot find file Neovim-v0.3.4.glibc2.17-x86_64.AppImage, triggering full download
/var/home/vcarvalho/Downloads/Neovim-v0.3.4.glibc2.17-x86_64.AppImage.part found, using as seed file
Target file: /var/home/vcarvalho/Downloads/Neovim-v0.3.4.glibc2.17-x86_64.AppImage
Reading seed file: /var/home/vcarvalho/Downloads/Neovim-v0.3.4.glibc2.17-x86_64.AppImage.part
Usable data from seed files: 0.000000%
Renaming temp file
Fetching remaining blocks
Downloading from https://github.com/neovim/neovim/releases/download/v0.3.4/Neovim-v0.3.4.glibc2.17-x86_64.AppImage

-------------------- 0.0%bad status code 404
Other error? -1
-1 returned
-------------------- 0.0% 0.0 kBps aborted    

failed to retrieve from Neovim-v0.3.4.glibc2.17-x86_64.AppImage, status -1
```
Am I missing something?

**Comment by justinmk** [comment]
https://github.com/neovim/neovim/blob/master/scripts/genappimage.sh#L86

By default, appimage barfs out a filename like `$APP-$VERSION.glibc$GLIBC_NEEDED-$ARCH.AppImage`. Our script renames that to `nvim.appimage`. But I think the zsync file doesn't expect this rename. I guess we need to do that somehow.

Not going to spend time on this, someone else needs to figure it out.

**Comment by justinmk** [comment]
@probonopd @TheAssassin we are renaming our appimage to `nvim.appimage`. How do we register this name-change with appimage so that `zsync`  understands it should look for 

    https://github.com/neovim/neovim/releases/download/v0.3.4/nvim.appimage

instead of:

    https://github.com/neovim/neovim/releases/download/v0.3.4/Neovim-v0.3.4.glibc2.17-x86_64.AppImage

?

**Comment by probonopd** [comment]
It is "documented" in bash:
https://github.com/AppImage/pkg2appimage/blob/3672c43909280c9538b8c2fca83be3de4d5de12b/functions.sh#L194-L252

Changing update information manually is not something our tools are supporting at the moment. Is there a urgent need for you to rename the file?

If you need more control over the AppImage generation, then maybe you want to invoke `appimagetool` [manually](https://github.com/AppImage/AppImageKit#appimagetool-usage) rather than using `generate_type2_appimage`. Then you are free to pass in custom [update information](https://github.com/AppImage/AppImageSpec/blob/master/draft.md#update-information) using `-u`.

**Comment by justinmk** [comment]
> . Is there a urgent need for you to rename the file?

- stable URLs 
- simpler docs
- simpler user experience

> If you need more control over the AppImage generation, then maybe you want to invoke `appimagetool` [manually](https://github.com/AppImage/AppImageKit#appimagetool-usage) rather than using `generate_type2_appimage`. Then you are free to pass in custom [update information](https://github.com/AppImage/AppImageSpec/blob/master/draft.md#update-information) using `-u`.

useful , thanks.

**Comment by dopessoa** [comment]
Hello, using `generate_type2_appimage` the header information on the zsync file is determined by the filename of the generated AppImage file. Take a look:

```
zsync: 0.6.2
> Filename: Neovim-v0.3.5.glibc2.17-x86_64.AppImage
MTime: Mon, 29 Apr 2019 20:17:33 +0000
Blocksize: 2048
Length: 10534856
Hash-Lengths: 2,2,5
> URL: Neovim-v0.3.5.glibc2.17-x86_64.AppImage
SHA-1: b565ad0d2309456a68861826e72d11b6694eeb26
```
My PR was a workaround changing these fields.

I couldn't find anything on AppImage docs referencing a way to change the values passed to `zsyncmake` using `generate_type2_appimage` to something akin to this:

`zsyncmake -f <filename> -u <url> <output_filename.zsync>`

The best solution really would be to find a way to set these fields on the zsync file using `appimagetool`.

**Comment by TheAssassin** [comment]
This script and all its contents are considered deprecated and horribly unmaintained. Rather switch to writing the 4-5 commands you really need by hand, instead of plugging together 3 functions from there.

Check out https://docs.appimage.org/packaging-guide/from-source/index.html for more information.

**Comment by dopessoa** [comment]
Sent a new PR #10027, this time doing things properly.

@TheAssassin, thanks for the documentation.

**Comment by justinmk** [comment]
can someone confirm that this is fixed for the `nightly` tag? (It won't be fixed for `v0.3.4` or any stable release until the next release)

**Comment by mnrce** [comment]
> can someone confirm that this is fixed for the `nightly` tag? (It won't be fixed for `v0.3.4` or any stable release until the next release)

I use AppImageUpdate to update nvim.appimage nightly and it works perfectly now.
